sdc/datatypes/hpat_pandas_series_rolling_functions.py

```diff
@@ -625,7 +618,70 @@ def hpat_pandas_series_rolling_sum(self):
     ty_checker = TypeChecker('Method rolling.sum().')
     ty_checker.check(self, SeriesRollingType)
 
-    return hpat_pandas_rolling_series_sum_impl
+    def _sdc_pandas_series_rolling_sum_impl(self):
+        win = self._window
+        minp = self._min_periods
+
+        input_series = self._data
+        input_arr = input_series._data
+        length = len(input_arr)
+        output_arr = numpy.empty(length, dtype=float64)
+
+        chunks = get_chunks(length)
+        for i in prange(len(chunks)):
```

Comment:
Does it helped? Can't wait to know the result 😄 

BTW you are not going to write all this monstrous code for every rolling function, don't you?

Reply:
I'm expecting to see generic implementation for the most of series methods something like this:
```python
windows = [WindowKind(window_size)]

for i in range(1, len(chunks)):
    windows.append(WindowKind(window_size))

for i in prange(len(chunks)):
    chunk = chunks[i]
    window = windows[i]
    interlude_start = max(0, chunk.start - window_size)
    interlude_stop = max(0, chunk.start)
    for j in range(interlude_start, interlude_stop):
        window.add(data, j)
    
    for j in range(chunk.start, chunk.stop)
        window.add(data, j)
        result[j] = window.get_result()
```

This is a pseudocode. You need to think about exact details